web/react/stores/preference_store.jsx

```diff
@@ -122,5 +122,6 @@ class PreferenceStoreClass extends EventEmitter {
 }
 
 const PreferenceStore = new PreferenceStoreClass();
+PreferenceStore.setMaxListeners(0);
```

Comment:
Could be, but the real problem is the data flow. If every component was listening to the stores the data flow would be a complete mess and all the components will then have dependencies on the stores (which is very un-React-like). If the data flow is always stores -> controllers -> children then it's very clean, maintainable, extendable and modular (which is very React-like).